src/components/layout/header/Header.tsx

```diff
@@ -78,20 +75,26 @@ const NAV_ITEMS = [
 ];
 
 const Header = (): JSX.Element => {
-    const dispatch = useAppDispatch();
     const [, setLocation] = useLocation();
 
-    const organizationId = useAppSelector(
-        (state) => state.me.info.organization_id
-    );
-    const channel = useAppSelector((state) => state.me.info.channel);
-    const userRole = useAppSelector((state) => state.me.info.user_role);
-    const username = useAppSelector((state) => state.me.info.user);
+    const {
+        info: {
+            organization_id: organizationId,
+            channel,
+            user_role: userRole,
+            user: username,
+        },
+        fetchLogout,
+    } = useAuthStore();
 
     const handleLogOut = () => {
-        dispatch(logOut())
-            .then(unwrapResult)
-            .then(() => setLocation(PATHS.LOGIN));
+        const logOut = async () => {
+            const logOut = await fetchLogout();
+            if (logOut !== null) {
```

Comment:
As for the login, I don't think there was a check on the success of the logout action, the setLocation was always performed. I think you can remove this check and remove the return value in fetchLogout.